### PR TITLE
JAMES-2488 Upgrade netflix curator & openjpa & jasypt(crypto)

### PR DESCRIPTION
**Pull request by trantienduchn**
```
com.netflix.curator:curator-client .................... 1.3.2 -> 1.3.3 GDF [zookeeper] Moved to org.apache.curator
com.netflix.curator:curator-framework ................. 1.3.2 -> 1.3.3 GDF [zookeeper] Moved to org.apache.curator
com.netflix.curator:curator-recipes ................... 1.3.2 -> 1.3.3 GDF [zookeeper] Moved to org.apache.curator
com.netflix.curator:curator-test ...................... 1.3.2 -> 1.3.3 GDF [zookeeper] Moved to org.apache.curator

org.apache.openjpa:openjpa ............................ 2.4.2 -> 3.0.0 MBA [db] 3.0.0 released one month ago, migration guide : http://openjpa.apache.org/builds/3.0.0/apache-openjpa/docs/main.html
org.jasypt:jasypt ..................................... 1.9.0 -> 1.9.2 MBA [crypto] use to store crypted messages into JPA, could probably be replaced by bouncycastle // low priority

```

### CODE REVIEW

File: mailbox/zoo-seq-provider/src/main/java/org/apache/james/mailbox/store/mail/ZooUidProvider.java [hunk 1]
```diff
@@ -99,6 +101,6 @@ public MessageUid nextUid(MailboxSession session, MailboxId mailboxId) throws Ma
     }
 
     public static <E extends MailboxId> String pathForMailbox(Mailbox mailbox) {
-        return mailbox.getMailboxId().serialize() + UID_PATH_SUFFIX;
+        return PATH_PREFIX + mailbox.getMailboxId().serialize() + UID_PATH_SUFFIX;
```
Comment (trantienduchn) [inline mailbox/zoo-seq-provider/src/main/java/org/apache/james/mailbox/store/mail/ZooUidProvider.java]: lastest curator requires a path with `/` at the beginning

File: mailbox/zoo-seq-provider/src/test/java/org/apache/james/mailbox/store/mail/ZooUidProviderTest.java [hunk 2]
```diff
@@ -137,73 +137,72 @@ public boolean equals(Object obj) {
 
     }
 
+    private static TestingServer testServer;
```
Comment (trantienduchn) [inline mailbox/zoo-seq-provider/src/test/java/org/apache/james/mailbox/store/mail/ZooUidProviderTest.java]: indentation fix, no code change.

**Comment by trantienduchn** [comment]
test this please

**Comment by chibenwa** [comment]
It's because `james-server-jpa-james-2488-c8d4ab0f-cf79-405d-9e41-d79c3e2adceb` did not start.... You might want to check manually that you can start it. For me it seems linked to the JPA upgrade.

File: mailbox/zoo-seq-provider/src/test/java/org/apache/james/mailbox/store/mail/ZooUidProviderTest.java (hunk 2 above)
Comment (rouazana) [reply mailbox/zoo-seq-provider/src/test/java/org/apache/james/mailbox/store/mail/ZooUidProviderTest.java]: next time, please try to do a separate commit for this

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
Gotcha ;-)

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**